symengine/polys/basic_conversions.h

```diff
@@ -10,14 +10,14 @@ namespace SymEngine
 // using `gen` as the genarator. Throws, if poly constructions not possible.
 // `ex` is the optional parameter for expanding the given `basic` or not.
 template <typename P>
-RCP<const P> from_basic(const RCP<const Basic> &basic,
-                        const RCP<const Basic> &gen, bool ex = false);
+RCP<const P> upoly_from_basic(const RCP<const Basic> &basic,
```

Comment:
I tried this (and `MSymEnginePoly` for the multivariate version, instead of `UPolyBase`)
```
template <typename T, typename P>
enable_if_t<std::is_base_of<UPolyBase<T, P>, P>::value, RCP<const P>>
from_basic(..)
```
but then, wouldn't this become a 2 template parameter functions? It's not able to deduce the "container" automatically